### ISSUE TITLE
some labels are not readable because of contrasts issue

### ISSUE THREAD

**Issue by karlcow** [webcompat/webcompat.com#1258]
I have the feeling that labels which are not handled by default have poor contrasts definition.

![capture d ecran 2016-12-26 a 10 09 10](https://cloud.githubusercontent.com/assets/505230/21473557/9547837a-cb53-11e6-9452-dd1a90d9be7f.png)

The white on grey here is not very good. :)

**Comment by lockettm** [comment]
I agree, the white on grey (#ededed) doesn't provide much contrast. A darker grey (#545454) would be more visible and meet accessibility standards.

**Comment by karlcow** [comment]
```html
<span class="js-Label-list">
  
    <span class="wc-Label wc-Label--badge js-Label" style="background-color:#ededed">
      firefox-mobile-(tablet)
    </span>
  
    <span class="wc-Label wc-Label--badge js-Label" style="background-color:#d4c5f9">
      worksforme
    </span>
  
  </span>
```

so the labels background colors are set directly on the element.

**Comment by magsout** [comment]
@karlcow 

>we fix all the #ededed on GitHub

IMHO, I think for the maintainability of the code, it's better to have the colors on Github, and tried to find a color that is well on Github and on webcompat. We have a different color of background.

**Comment by zoepage** [comment]
At the SF work week, we spoke about the labels in general and realized we use too many colors. The decision to clean up the labels was made. 

1. We group labels by type. E.g. browsers, devices or status
2. All groups get a color assigned.

##### List of all used labels (last state: 07.17) and colors (if not default)
<details>Context Labels
Default: #ED
Browser-android #2AB
Browser-chrome: #E65
Browser-edge: #EC8
Browser-Firefox-mobile #FC0 (Status-needs-info)
Browser-Firefox #F40
Browser-Iceweasel #05C
Browser-IE #0BE
Browser-opera #A00
Browser-safari #29F
Browser-ucweb #FCC 
Ios #0D2
Mobile #090 (same as status-fixed)
NSFW #C00
Status-contactready #BEB
Status-duplicate #333
Status-fixed #090 (same as mobile)
Status-ie-bugbar #BDD (same as Type-webvr)
Status-incomplete #D30
Status-invalid #E6
Status-needscontact #E12
Status-needsdiagnosis-important #F86 (same as Status-needsdiagnosis)
Status-needsdiagnosis #F86 (same as Status-needsdiagnosis-important)
Status-needs-info #FC0 (same as Browser-Firefox-mobile)
Status-needs-triage #F90
Status-sitewait #077
Status-wontfix #000
Status-worksforme #DCF
Type-cssrule #51E
Type-flash #B00
Type-media #FDC
Type-webp #E99
Type-webrender #33
Type-webvr #BDD (same as Status-ie-bugbar)
Browser-all
Browser-blackberry-browser
Browser-blackberry
Browser-chrome-mobile-ios
Browser-chrome-mobile
Browser-chromium
Browser-edge-mobile
Browser-fennec
Browser-firefox-aurora-mobile
Browser-firefox-esr
Browser-firefox-focus
Browser-firefox-for-android
Browser-firefox-for-ios
Browser-firefox-ios
browser-firefox-mobile-(tablet)
Browser-firefox-mobile-nightly
Browser-hsafari
Browser-ie-mobile
Browser-ie11
Browser-lynx
Browser-midori
Browser-mobile-safari
browser-myie-mobile-(tablet)
Browser-myie-mobile
Browser-myie
Browser-nightly
Browser-opera-desktop
Browser-opera-mini-v
Browser-opera-mini
Browser-puffin-browser-android
Browser-seamonkey
Browser-uc-browser
Browser-vivaldi
Browser-webkit-nightly
Browser-yandex-browser
Os-android
Windows
</details>

**Comment by karlcow** [comment]
Thanks @zoepage 

Some other constraints. 
* Status labels will disappear soon.
* labels colors are currently set through github backend. We could change that by setting them through a configuration file and ignore the labels on GitHub. (or make a script setting the colors on GitHub backend)
* we currently have:
  * `type-*`
  * `browser-*`
  * `status-*`
  * and the rest not really categorized, probably including `os-*`

I wonder if we could improve a bit more the browser labeler to avoid too much variability. We have things like `Browser-firefox-for-ios` and `Browser-firefox-ios` for example, where it could have been. `browser-firefox` and `os-ios`. That offers more flexibility for search filtering.

**Comment by zoepage** [comment]
> Status labels will disappear soon.

🎉

> We could change that by setting them through a configuration file and ignore the labels on GitHub.

Changing them in the GH interface makes sense, so we stay transparent and do not create "magical behavior" people do not know how to e.g. change. It is also nice to give it some contingency in both interfaces as people sometimes work in GH issues (although they should work on webcompat.com).

> we currently have...
We also have things like NSFW. Where would this go?

> I wonder if we could improve a bit more the browser labeler to avoid too much variability. We have things like Browser-firefox-for-ios and Browser-firefox-ios for example, where it could have been. browser-firefox and os-ios. That offers more flexibility for search filtering.

++1

**Comment by karlcow** [comment]
@zoepage 
We could keep the consistency in between webcompat.com and github by pushing the colors to Github. see https://developer.github.com/v3/issues/labels/#update-a-label

The benefits is that it makes it possible to have a nice file with the config for the colors without going through the UI to set the colors manually.

Would that work for you?

**Comment by karlcow** [comment]
@zoepage 

> We also have things like NSFW. Where would this go?

It could go in type as we can have multiple `type-*` for an issue. Or it could just stay as-is.

**Comment by zoepage** [comment]
@karlcow that sounds like a good idea :) I like configs in one place :)

Also the type-NSFW sounds good :)

**Comment by zoepage** [comment]
As a new config was added for label-colors (YAAY!) in #1754 we could move on with cleaning up the colors to fix the contrast issue as addressed here: https://github.com/webcompat/webcompat.com/issues/1258#issuecomment-319061156 and also clean up the labels itself? (If this already happened and I missed this, sorry). What do you think @karlcow @miketaylr @MDTsai ?

Thanks!

**Comment by karlcow** [comment]
@zoepage Do you want to do that?
This is a tool which will help
https://webaim.org/resources/contrastchecker/

**Comment by zoepage** [comment]
@karlcow 
> Do you want to do that?

You mean cleaning up the labels and introducing a system for that as explained in here https://github.com/webcompat/webcompat.com/issues/1258#issuecomment-319061156 ? This is what needs to happen before we can tweak the colors. Happy to add that color tweak to our community sprint very next week.

**Comment by zoepage** [comment]
This has been adressed in design/#100

**Comment by karlcow** [comment]
Probably https://github.com/webcompat/design/issues/100
